extras/docker/fromsource/heketi-start.sh

```diff
@@ -1,19 +1,67 @@
 #!/bin/sh
 
-if [ -f /backupdb/heketi.db.gz ] ; then
-    gunzip -c /backupdb/heketi.db.gz > /var/lib/heketi/heketi.db
-    if [ $? -ne 0 ] ; then
-        echo "Unable to copy database"
+: ${HEKETI_PATH:=/var/lib/heketi}
+: ${BACKUPDB_PATH:=/backupdb}
+
+echo "Setting up heketi database"
+
+if [[ -d "${HEKETI_PATH}" ]]; then
+    # Test that our volume is writable.
+    touch "${HEKETI_PATH}/test" && rm "${HEKETI_PATH}/test"
+    if [ $? -ne 0 ]; then
+        echo "${HEKETI_PATH} is read-only"
         exit 1
     fi
-    echo "Copied backup db to /var/lib/heketi/heketi.db"
-elif [ -f /backupdb/heketi.db ] ; then
-    cp /backupdb/heketi.db /var/lib/heketi/heketi.db
-    if [ $? -ne 0 ] ; then
-        echo "Unable to copy database"
+
+    if [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; then
+        echo "No database file found" | tee "${HEKETI_PATH}/container.log"
+        out=$(mount | grep "${HEKETI_PATH}" | grep heketidbstorage)
+        if [[ $? -eq 0 ]]; then
+            echo "Database volume found: ${out}" | tee "${HEKETI_PATH}/container.log"
+            echo "Database file is expected, waiting..." | tee "${HEKETI_PATH}/container.log"
+            check=0
+            while [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; do
+                sleep 5
+                if [[ ${check} -eq 5 ]]; then
+                   echo "Database file did not appear, exiting." | tee "${HEKETI_PATH}/container.log"
+                   exit 1
+                fi
+                ((check+=1))
+            done
+        fi
+    fi
+
+    stat "${HEKETI_PATH}/heketi.db" | tee "${HEKETI_PATH}/container.log"
+    # Workaround for scenario where a lock on the heketi.db has not been
+    # released.
+    flock -w 60 "${HEKETI_PATH}/heketi.db" true
+    if [[ $? -ne 0 ]]; then
+        echo "Database file is read-only" | tee "${HEKETI_PATH}/container.log"
+    fi
```

Comment:
I would still prefer that we exit instead of leaving heketi service in read only.

Reply:
I take back this comment. Probably a better method is to try reopening the db in read-write mode during heketi operation. Keep the code as is.